ibis/backends/sql/compilers/pyspark.py

```diff
@@ -683,5 +689,8 @@ def visit_ArraySum(self, op, *, arg):
     def visit_ArrayMean(self, op, *, arg):
         return self._array_reduction(dtype=op.dtype, arg=arg, output=operator.truediv)
 
+    def visit_ToJSONArray(self, op, *, arg):
+        return self.f.from_json(arg, sge.Literal.string("array<string>"))
```

Comment:
```suggestion
        return arg
```

I *think* this should probably do nothing, since there's no official JSON type in Spark.

Reply:
But if we do this, the following code will throw an error in spark. 🤔
```python
tab = backend.table("table") 
print(tab.select(tab.val.array[0].name("value").compile())
```

result:
```
SELECT ELEMENT_AT(`t0`.`val`, 0 + 1) AS `value` from `table`;
```

Reply:
It looks like the [variant type is landing in PySpark 4.0.0,](https://spark.apache.org/docs/4.0.0-preview2/api/python/reference/pyspark.sql/api/pyspark.sql.types.VariantType.html) which will solve the problem of having a type to represent JSON-like data. I'm not sure it's worth working around this problem for earlier Spark versions.

Reply:
So, i tried the `variant` type in the spark `4.0.0-preview2`. it actually works great with ibis's current implement.

```sql
SELECT CAST(FROM_JSON(a.a, 'variant') as ARRAY<STRUCT<A: STRING>>).A FROM (
    SELECT '[{"A":"B"}]' as a
) a
```
It outputs (amazingly) 
```
[B]
```

But I think it's too early to support spark 4.0.0 now.

So. I have another propose.

**Another Thoughts**
We create a custom function class names `FromJson`, and we check the input type is `FromJson` or not. If that is, we extract the first parameters from it and cast the input to the target type.

Snippets:
```python3
class FromJson(sge.Func):
    arg_types = {"expressions": True}
    is_var_len_args = True


class PySparkCompiler(SQLGlotCompiler):
    # ...
    def visit_ToJSONArray(self, op, *, arg):
        return FromJson.from_arg_list([arg, sge.Literal.string("array<string>")])

    def visit_ToJsonMap(self, op, *, arg):
        return FromJson.from_arg_list([arg, sge.Literal.string("map<string, string>")])

    def visit_Cast(self, op, *, arg, to):
        if isinstance(arg, FromJson):
            return FromJson.from_arg_list(
                [
                    arg.expressions[0],
                    sge.Literal.string(PySparkTypeMapper.from_ibis(to).simpleString()),
                ]
            )
        # ...
```
What do you think?